skimage/restoration/_denoise.py

```diff
@@ -277,17 +282,38 @@ def denoise_tv_bregman(image, weight, max_iter=100, eps=1e-3, isotropic=True):
 
     """
     image = np.atleast_3d(img_as_float(image))
-    image = np.ascontiguousarray(image)
 
-    rows = image.shape[0]
-    cols = image.shape[1]
-    dims = image.shape[2]
-
-    shape_ext = (rows + 2, cols + 2, dims)
+    shape_ext = np.asarray(image.shape)
+    shape_ext[0:2] += 2
 
     out = np.zeros(shape_ext, image.dtype)
-    _denoise_tv_bregman(image, image.dtype.type(weight), max_iter, eps,
-                        isotropic, out)
+
+    if multichannel:
+        for c in range(image.shape[-1]):
+            if np.ndim(image) == 3:
+                channel_in = np.ascontiguousarray(image[..., c, None])
+                channel_out = np.ascontiguousarray(out[..., c, None])
+
+                _denoise_tv_bregman(channel_in, image.dtype.type(weight),
+                                    max_iter, eps, isotropic, channel_out)
+
+                out[..., c] = channel_out[..., -1]
+
+            else:
+                channel_in = np.ascontiguousarray(image[..., c])
```

Comment:
this only works if the user provides you with a 4D image. is this something that you are considering?

The algorithm seems to assume that we have a 2D + color image. therefore, i'm not sure how the dimension "2" of the image relates to the real data.

Reply:
Compare latest commit